openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -309,8 +314,98 @@ def _upgrade_reminder_schedules_for_bin(site, target_day, bin_num, org_list, exc
         yield (user, first_schedule.enrollment.course.language, template_context)
 
 
+class CourseUpdate(ScheduleMessageType):
+    pass
+
+
+@task(ignore_result=True, routing_key=ROUTING_KEY)
+def course_update_schedule_bin(
+    site_id, target_day_str, day_offset, bin_num, org_list, exclude_orgs=False, override_recipient_email=None,
+):
+    target_day = deserialize(target_day_str)
+    msg_type = CourseUpdate()
+
+    for (user, language, context) in _course_update_schedules_for_bin(
+        Site.objects.get(id=site_id),
+        target_day,
+        day_offset,
+        bin_num,
+        org_list,
+        exclude_orgs
+    ):
+        msg = msg_type.personalize(
+            Recipient(
+                user.username,
+                override_recipient_email or user.email,
+            ),
+            language,
+            context,
+        )
+        _course_update_schedule_send.apply_async((site_id, str(msg)), retry=False)
+
+
+@task(ignore_result=True, routing_key=ROUTING_KEY)
+def _course_update_schedule_send(site_id, msg_str):
+    site = Site.objects.get(pk=site_id)
+    if not ScheduleConfig.current(site).deliver_course_update:
+        return
+
+    msg = Message.from_string(msg_str)
+    ace.send(msg)
+
+
+def _course_update_schedules_for_bin(site, target_day, day_offset, bin_num, org_list, exclude_orgs=False):
+    week_num = abs(day_offset) / 7
+    beginning_of_day = target_day.replace(hour=0, minute=0, second=0)
+    schedules = get_schedules_with_target_date_by_bin_and_orgs(
+        schedule_date_field='start',
+        target_date=beginning_of_day,
+        bin_num=bin_num,
+        num_bins=COURSE_UPDATE_NUM_BINS,
+        org_list=org_list,
+        exclude_orgs=exclude_orgs,
+        order_by='enrollment__course',
+    )
+
+    LOG.debug('Course Update: Query = %r', schedules.query.sql_with_params())
+
+    for schedule in schedules:
+        enrollment = schedule.enrollment
+        try:
+            week_summary = get_course_week_summary(enrollment.course_id, week_num)
+        except CourseUpdateDoesNotExist:
+            continue
```

Comment:
Checking my understanding: We expect `CourseUpdateDoesNotExist` to happen a lot, right?

Reply:
Initially, yes.  Eventually, less.
I'm also unsure of what the Course Module will eventually be.  So if it raises an exception from that layer, it could get propagated up to this point - and not need to be intercepted checking for `None` values along the way.  I'm definitely open to changing this - as we figure out the underlying data-model side of it.